### PR TITLE
Fixes problems related to tax calculation and price filter in multistore setup

### LINKED ISSUE #4376: Parts of code not checking for multistore
## Current behavior
```
TypeError: Cannot read property 'currencySign' of undefined
```
https://github.com/DivanteLtd/vue-storefront/blob/v1.11.3/core/modules/catalog-next/store/category/getters.ts#L80-L81

Additional problems in:
 - `core/modules/catalog/store/tax/getters.ts`
 - `core/modules/catalog/store/tax/actions.ts`

## Repository
`v1.11.3`

## Can you handle fixing this bug by yourself?

- [x] YES
- [ ] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [ ] This is a bug report for test version on https://test.storefrontcloud.io - In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] This is a bug report for current Release Candidate version on https://next.storefrontcloud.io - In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [x] This is a bug report for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version hotfix - In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.

### PR DESCRIPTION
**Pull request by juho-jaakkola**
### Related Issues

closes #4376 

### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [x] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: core/lib/types.ts [hunk 1]
```diff
@@ -21,6 +21,8 @@ export interface StoreView {
   },
   tax: {
     sourcePriceIncludesTax: boolean,
+    finalPriceIncludesTax: boolean,
+    deprecatedPriceFieldsSupport: boolean,
```
Comment (gibkigonzo) [inline core/lib/types.ts:25]: please change all of this as optional 
```
    sourcePriceIncludesTax?: boolean,
    finalPriceIncludesTax?: boolean,
    deprecatedPriceFieldsSupport?: boolean,
```

Comment (juho-jaakkola) [reply core/lib/types.ts:25]: ok, will do

**Comment by gibkigonzo** [comment]
@juho-jaakkola thanks for this, please update changelog

**Comment by juho-jaakkola** [comment]
@gibkigonzo I'm not completely sure what to put into the changelog.

Do you think the text in the commit message would be enough?

**Comment by juho-jaakkola** [comment]
- [x] Marked the properties in `core/lib/types.ts` as optional
- [x] Updated the CHANGELOG

**Pull request merged**